### PR TITLE
Refactor to remove Promise.bind()

### PR DESCRIPTION
**Pull request by sleepwithcoffee**
Addresses: #8369

### CODE REVIEW

File: lib/plugins/plugin/search.js [hunk 1]
```diff
@@ -23,12 +22,12 @@ class PluginSearch {
...
 
     this.hooks = {
-      'plugin:search:search': async () => BbPromise.bind(this).then(this.search),
+      'plugin:search:search': async () => this.search(),
     };
   }
 
   async search() {
-    return BbPromise.bind(this)
+    return Promise.resolve(this)
```
Comment (medikoo) [inline lib/plugins/plugin/search.js:30]: Same here

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @sleepwithcoffee, it looks very good. I have just one improvement suggestion

File: lib/plugins/plugin/list.js [hunk 2]
```diff
@@ -22,13 +21,13 @@ class PluginList {
...
     this.hooks = {
-      'plugin:list:list': async () => BbPromise.bind(this).then(this.list),
+      'plugin:list:list': async () => this.list(),
     };
   }
 
   async list() {
-    return BbPromise.bind(this)
-      .then(this.getPlugins)
+    return Promise.resolve(this)
```
[the remark this answers is not in the document]
Comment (sleepwithcoffee) [reply lib/plugins/plugin/list.js:29]: We still need that to pass `this` context to the following functions, either that or:

```
return this.getPlugins()
      .then(this.display(plugins).bind(this));
```

[the remark this answers is not in the document]
Comment (medikoo) [reply lib/plugins/plugin/list.js:29]: It's not needed, you can write same as:

```javascript
async () {
  return this.getPlugins().then((plugins) => this.display(plugins));
}
```

and actually we can already refactor to async await as:

```javascript
async  list() {
  await this.display(await this.getPlugins());
}
```

[the remark this answers is not in the document]
Comment (sleepwithcoffee) [reply lib/plugins/plugin/list.js]: thanks for pointing it out, it was so obvious that I didn't think about it

File: lib/plugins/plugin/search.js (hunk 1 above)
Comment (sleepwithcoffee) [reply lib/plugins/plugin/search.js]: done

**Review by medikoo — APPROVED** [review]
Looks great 👍 Thank you!

**Pull request merged**